### PR TITLE
feat: adds reasons to models

### PR DESCRIPTION
**Pull request by andivido**
Adds the reason string/ withReason function to the relevant models, allowing the SDK to send the Merchant API a reason, which should allow us to create cancellations/refunds for Novuna via the SDK

### CODE REVIEW

File: lib/Models/ApplicationCancellation.php [hunk 1]
```diff
@@ -96,4 +101,20 @@ public function withOrderItems(array $orderItems)
...
     }
+
+    /**
+     * With reason.
+     *
+     * @param string $reason
+     *
+     * @return \Divido\MerchantSDK\Models\ApplicationCancellation
+     */
+    public function withReason($reason)
```
Comment (simeon-anastasov-divido) [inline lib/Models/ApplicationCancellation.php:112]: @andivido I'm not familiar with the style of the repo (perhaps you're following that) and I also see that there are no strict type checks but yeah - maybe add a type check or cast the parameter to a string? Or it is done automatically somewhere down the line?

**Review by simeon-anastasov-divido — APPROVED** [review]

File: lib/Models/ApplicationCancellation.php (hunk 1 above)
Comment (andivido) [reply lib/Models/ApplicationCancellation.php:112]: I think the SDK is kept as compatible as possible with as low a version of PHP as possible, for the benefit of merchants trying to implement it (though we do seem to [type check](https://github.com/dividohq/merchant-api-pub-sdk-php/blob/7394f008bb40eaffcc7a982863334f355633782b/lib/Models/ApplicationCancellation.php#L96) occassionally). We should maybe expect clients to have at least the minimum supported version of PHP on their systems though, I guess. It's a separate task that needs to be carried out throughout the SDK though

Comment (ian-lewis-d) [reply lib/Models/ApplicationCancellation.php:112]: @andivido Where are the Merchant Reasons defined?

Comment (andivido) [reply lib/Models/ApplicationCancellation.php:112]: Jeepers, lets see: This connects to the Merchant API Pub: No enums [here](https://github.com/dividohq/merchant-api-pub/blob/master/src/Divido/JsonSchemas/refund_store.json), so let's check upstream in the application-api: Seems like we're still expecting a string (though in the form of a const) [here](https://github.com/dividohq/application-api/blob/master/src/Divido/JsonSchemas/post_refund.json). Ah, [here](https://github.com/dividohq/application-api/blob/3f2e84dd2f58381967b5973cb1e4280680d44304/src/Divido/Services/Refund/RefundService.php#L46) we are, and [here](https://github.com/dividohq/application-api/blob/3f2e84dd2f58381967b5973cb1e4280680d44304/src/Divido/Services/Refund/RefundService.php#L98)'s the validation check. Interesting to note if the lender doesn't require a reason, it'll throw an error if one is supplied

Comment (simeon-anastasov-divido) [reply lib/Models/ApplicationCancellation.php:112]: well I guess that's a safety net for the rare occasions where people overshare? I believe there was a good although hacky reason why these are not defined at the request validation level. @USilviuDvdo any thoughts? It was lender-specific and not universal across?

Comment (USilviuDvdo) [reply lib/Models/ApplicationCancellation.php:112]: @simeon-anastasov-divido Indeed, only Novuna has reasons from what I know so far.

**Review by USilviuDvdo — APPROVED** [review]

File: lib/Models/ApplicationCancellation.php (hunk 1 above)
Comment (andivido) [reply lib/Models/ApplicationCancellation.php:112]: Also interestingly, conversely, cancellation reasons [are allowed](https://github.com/dividohq/application-api/blob/279018465920fb00b322001e20b35f414d6c005b/src/Divido/Services/Cancellation/CancellationService.php#L82) if the lender doesn't require reasons. We should probably be putting Refund validation through the same [RefundaAndCancellationReason Helper](https://github.com/dividohq/application-api/blob/master/src/Divido/Helpers/RefundAndCancellationReason.php) class

Comment (simeon-anastasov-divido) [reply lib/Models/ApplicationCancellation.php:112]: @andivido ah, fair spot - @USilviuDvdo perhaps having consistency across is not a bad idea :)

Comment (andivido) [reply lib/Models/ApplicationCancellation.php:112]: This whole can of worms is all @ian-lewis-d 's fault!

**Pull request merged**